src/handlers/payout/action.ts

```diff
@@ -194,6 +194,17 @@ export const incentivesCalculation = async (): Promise<IncentivesCalculationResu
     throw new Error(`Permit generation disabled because this issue didn't qualify as bounty.`);
   }
 
+  // check for label altering here
+  const labelChanges = await getLabelChanges(repository.full_name, [issueDetailed.priceLabel, issueDetailed.priorityLabel, issueDetailed.timelabel]);
+
+  if (labelChanges) {
+    // if approved is still false, it means user was certainly not authorized for that edit
+    if (!labelChanges.approved) {
+      logger.info(`Skipping... label was changed by unauthorized user`);
+      throw new Error(`Permit generation disabled because this label: ${labelChanges.label_to} on this issue was edited by an authorized user`);
```

Comment:
```suggestion
      throw new Error(`! Permit generation disabled because label "${labelChanges.label_to}" was modified by an unauthorized user`);
```